lib/build/bundle.js

```diff
@@ -220,15 +244,25 @@ exports.Bundle = class {
       if (includeBundles) {
         bundleMetadata[currentBundle.moduleId] = currentBundle.getBundledModuleIds();
       }
-
-      config.paths[currentBundle.moduleId] = location + '/' + currentBundle.moduleId;
+      //If build revisions are enabled, append the revision hash to the appropriate module id
+      config.paths[currentBundle.moduleId] = location + '/' + currentBundle.moduleId + (buildOptions.rev && currentBundle.hash ? '-' + currentBundle.hash : '');
     }
 
     if (includeBundles) {
       config.bundles = bundleMetadata;
     }
 
-    return 'requirejs.config(' + JSON.stringify(config) + ');'
+    return 'requirejs.config(' + JSON.stringify(config) + ')'
+  }
+  
+  setIndexFileConfigTarget(platform, location){
+    //Replace the reference to the vendor bundle in the "index.html" file to use the correct build revision file (or remove the build revision hash);
+    let indexLocation = platform.index;
+    let data = fs.readFileSync(indexLocation);
+    let outputDir = platform.baseUrl || platform.output; //If we have a baseUrl, then the files are served from there, else it's the output
+    let configMatcher = new RegExp(`\\b(?:src=("|')(.*))(${outputDir}(\/|\\\\)${this.moduleId}(.*?).js)(?:("|'))`);
```

Comment:
I think you'll need to escape `outputDir` and `moduleId` here, in case they contain slashes (subdirs, scoped modules) or other characters that need to be escaped. I recommend using https://github.com/sindresorhus/escape-string-regexp for this job.

It would be good to add some tests for this method specifically, as I'm not sure what's it supposed to do exactly without digging too deep into the package. Maybe you could refactor the Regex + match-replacement part of this method into a separate, pure function outside of the class. That way you could just create 2-3 simple tests: feeding it some sample text and comparing with desired results? 
Should be relatively, easy to do and will make this review much easier :)